yarn.lock

```diff
@@ -6750,7 +6306,7 @@ metro-react-native-babel-preset@0.72.1:
     "@babel/template" "^7.0.0"
     react-refresh "^0.4.0"
 
-metro-react-native-babel-transformer@0.72.1, metro-react-native-babel-transformer@^0.72.1:
+metro-react-native-babel-transformer@^0.72.1:
   version "0.72.1"
```

Comment:
Unfortunately I'm not sure this solves the "multiple versions" problem, because you get issues like this where Yarn won't update if it has an existing entry that satisfies the semver range for a given dependency - there are a mix of 0.72.3 and 0.72.1 in this resulting lockfile - and app developers will get the same issue in their project lockfiles.

I *think* the only way to ensure all the Metro packages to reliably stay aligned is for everything that depends on them to pin them exactly, but I'm open to suggestions..

Reply:
ok - I just did what @thymikee recommended to me as bumps; I always tend do prefer fixed versions. I assumed that was an agreement between CLI and Metro?

Mike, what should we do?

Reply:
I think you may be right @robhogan, I recall situations like this. Although I never really bothered as an app dev, because I edited the lockfile or ran yarn-deduplicate. As a library author I prefer using `^` or `~` instead of fix versions, so that transitive deps can "fix themselves" without me issuing a new version. But, it looks like in RN repo Metro is always fixed to a specific version. I'm good with doing the same for the CLI, if we agree that's the way it should be. 

If no-one is opposed to this, I'll issue CLI 9.1.3 with Metro fixed to 0.72.3 (or whatever is the latest) tomorrow. Then this PR could be updated to 0.72.3 as well, to get the latest fixes :)

Reply:
(I might've missed some discussion here so apologies if I'm contradicting some other conclusion, but yeah I think it's clear from the lockfile that this won't necessarily help - Yarn is conservative by design and won't necessarily update everything in step when a lockfile is present.)

A big part of _why_ this multiple versions thing is actually a problem is because Metro uses dynamic `require` calls to provide configurable/pluggable bits of functionality, eg `const transformer = require(babelTransformerPath)`. Those "dependencies" aren't necessarily explicit in the `package.json` of the requiring package, and unless they're absolute paths there's no real telling what they'll resolve to, which is how we get invocations crossing version boundaries within Metro even though all of Metro's packages use exact pinned dependencies. I think solving that would mean moving from configuration by dynamic `require`, to configuration by code injection, and (strictly IMO) we should fix all of that in Metro.

In the mean time, I don't see an alternative other than exact pinning everywhere and making sure they're kept consistent.

Reply:
> If no-one is opposed to this, I'll issue CLI 9.1.3 with Metro fixed to 0.72.3 (or whatever is the latest) tomorrow.

FYI, we'll be releasing 0.73.0 tomorrow or Thursday with a couple of smallish breaking changes
 - Default minifier Uglify-ES => Terser for better syntax support (bigint-friendly exponentiation operator, numeric separators)
 - Drop support for old Watchman versions (if anyone cares about this 😅)

I've no preference on whether we wait for that, but I guess it'd be a major bump for the CLI. @cortinico, @Dmitryrykun?

Reply:
@robhogan let's handle that separately - I want to be able to cherry pick this into the 0.70 release

Reply:
For context, there was a brief discussion of exact vs caret dependencies in `#releases-coordination` on Discord, where @thymikee and myself were in favour of caret dependencies: https://discord.com/channels/514829729862516747/514832743654228009/1022065446305923073

Based on @robhogan's comments it does seem that exact deps on Metro would be more reliable for our users. But... doesn't it mean we would also need the RN-->CLI dependency to be exact? (Otherwise the CLI can't safely upgrade its pinned version of Metro in a patch version)

Reply:
https://github.com/react-native-community/cli/pull/1705

Reply:
Released v9.1.3